### ISSUE TITLE
Cake does not find SYSWOW64-mapped NuGet for installing addins

### ISSUE THREAD

**Issue by hurcane** [cake-build/cake#1351]
### What You Are Seeing?
Cake is finding NuGet.exe via PATH environment variable during preparation, but not seeing it when installing addins.
### What is Expected?
Cake should find the executable that is in the path.
### What version of Cake are you using?
0.1.6.2
### Are you running on a 32 or 64 bit system?
64-bit system, but running X86 version of PowerShell.
### What environment are you running on?  Windows? Linux? Mac?
Windows
### Are you running on a CI Server?  If so, which one?
No.
### How Did You Get This To Happen? (Steps to Reproduce)
Cake file is very simple:
#addin nuget:?package=Cake.MSBuildTask
var target = Argument("target", "Default");
Task("BuildTDAO")
	.Does(() =>
{
	Information("TDAO build here");
});
Task("Default")
	.Does(() =>
{
	Information("TDAO build here");
});
RunTarget(target);

NuGet is physically located in c:\Windows\SYSWOW64 folder. It is in the path for 32-bit executable, but not for 64-bit executables.

When the build is run from the 32-bit version of PowerShell, the preparation steps are finding NuGet.exe on the path. However, the Installing AddIns phase is not finding it.
### Output Log
~~~sh
PS C:\Users\dvb.TRIBMAIL\Documents\TrulinXDevTrunk> .\build.ps1 -verbose
Preparing to run build script...
VERBOSE: Trying to find nuget.exe in PATH...
VERBOSE: Found in PATH at C:\Windows\system32\nuget.exe.
VERBOSE: Restoring tools from NuGet...
VERBOSE: Feeds used:
  C:\Users\dvb.TRIBMAIL\AppData\Local\NuGet\Cache
  C:\Users\dvb.TRIBMAIL\.nuget\packages\
  \\trudev2\OzServerInstall\NuGetPackages
  https://www.nuget.org/api/v2/
  C:\Program Files (x86)\Microsoft SDKs\NuGetPackages\

All packages listed in C:\Users\dvb.TRIBMAIL\Documents\TrulinXDevTrunk\tools\packages.config are already installed.
Running build script...
Analyzing build script...
Processing build script...
Installing addins...
Error: Could not locate nuget.exe.
~~~

**Comment by devlead** [comment]
Do you have the same issue with v0.17.0 release?

Does it work if you in your `build.ps1` add `c:\Windows\SYSWOW64` to path?

One way so ensure it'll always work would probably be to select a version from [nuget downloads](https://dist.nuget.org/index.html) and pin to that version(i.e. https://dist.nuget.org/win-x86-commandline/v3.5.0/NuGet.exe) and download if it doesn't exists in repo tools folder download it.

**Comment by hurcane** [comment]
> Do you have the same issue with v0.17.0 release?
Yes.

> Does it work if you in your build.ps1 add c:\Windows\SYSWOW64 to path?
Adding $Env:Path += ";C:\Windows\SYSWOW64" just before invoking cake makes it work.

For my situation, I resolved the issue by simply deleting the nuget.exe that was in the C:\Windows\SYSWOW folder. That was an older version from when I was first experimenting in 2015 with adding NuGet packages to a Visual Studio solution.

Having dealt with x86 vs. x64 issues numerous times myself, I know that the issue is that Cake.exe ran as an x64 process, and the x64 environment would not include the SYSWOW64 folder in the path. The PowerShell script was running as an x86 process, which did include the SYSWOW64 (as System32).

I did a test compile of Cake for the x86 platform, and that also made my original scenario work. However, this is such an edge case situation that I don't think it is worth making an X86 build available for download.
